AdvancedVehicleOptions/AdvancedVehicleOptions.cs

```diff
@@ -373,8 +373,8 @@ public static void InitConfig()
             DefaultOptions.CheckForConflicts();
 
             // Update existing vehicles
-            new EnumerableActionThread(VehicleOptions.UpdateCapacityUnits);
-            new EnumerableActionThread(VehicleOptions.UpdateBackEngines);
+            SimulationManager.instance.AddAction(VehicleOptions.UpdateCapacityUnits);
+            SimulationManager.instance.AddAction(VehicleOptions.UpdateBackEngines);

```

Comment:
We are in InitConfig method.
Question is: do you case about order of operations?

I'm asking because now and also previously those actions were running fully async (non-blocking) so they most likely didn't finish work before stepping to the next instruction on the **line 379**